client/src/components/AvailableRoles.js

```diff
@@ -2,95 +2,301 @@
+import RemoveIcon from "@material-ui/icons/Remove";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Divider from "@material-ui/core/Divider";
+import InsertLinkIcon from "@material-ui/icons/InsertLink";
+import { useUserState } from "../ContextProvider/user";
+import io from "socket.io-client";
+//import { MatchContext } from "../ContextProvider/match";
 
 const useStyles = makeStyles({
     grid: {
-        marginLeft: '100px',
-        marginBottom: '30px',
-        width: '100%',
-        border: '1px solid theme.palette.divider',
-        borderRadius: 'theme.shape.borderRadius',
-        backgroundColor: 'theme.palette.background.paper',
+        marginLeft: "100px",
+        marginBottom: "30px",
+        width: "100%",
+        border: "1px solid theme.palette.divider",
+        borderRadius: "theme.shape.borderRadius",
+        backgroundColor: "theme.palette.background.paper",
     },
     list: {
-        width: '50%',
-        marginLeft: '140px',
-        marginBottom: '20px'
+        width: "50%",
+        marginLeft: "140px",
+        marginBottom: "20px",
     },
     listItem: {
-        marginLeft: '-10px'
+        marginLeft: "-10px",
     },
     sectionOne: {
-        marginRight: '20%'
+        marginRight: "20%",
     },
     sectionTwo: {
-        marginLeft: '5%'
+        marginLeft: "5%",
     },
     submitButton: {
-        marginLeft: '35%'
-    }
-  });
+        marginLeft: "35%",
+    },
+    block: {
+        display: "block",
+    },
+    blueText: {
+        color: "#5B86E5",
+    },
+    redText: {
+        color: "#ff5e62",
+    },
+    xsmall: {
+        height: "30px",
+        width: "30px",
+        boxShadow: "0px 3px 24px -8px black",
+        verticalAlign: "middle",
+        backgroundColor: "lightgray",
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+        marginLeft: "10px",
+        "&:hover": {
+            backgroundColor: "gray",
+        },
+    },
+    alignCenter: {
+        justifyContent: "center",
+    },
+});
 
 export default function AvailableRoles() {
     const classes = useStyles();
+    const { user } = useUserState();
+    // const { matchState, setMatchState } = useContext(MatchContext); needs match context provider
+    const socketRef = useRef();
+
+    useEffect(() => {
+        socketRef.current = io.connect("localhost:3002/"); //will change when the other stuff gets approved
+        socketRef.current.on("update-game-engine", (game) => {
+            //setMatchState(game);
+        });
+        return () => socketRef.current.disconnect();
+    }, [setMatchState]);
+
+    const assignRole = (role) => {
+        // Check if the current user is not assigned a role.
+        if (
+            [
+                ...matchState.redGuessers,
+                ...matchState.blueGuesser,
```

Comment:
might want to make the names consistent

Reply:
Yeah this was removed though because I didn't need this section. It's showing "Outdated" on github and its not in the repo but still shows for you?